fido2/server.py

```diff
@@ -186,8 +190,8 @@ def register_begin(
         )
 
         return (
-            {
-                "publicKey": PublicKeyCredentialCreationOptions(
+            CredentialCreationOptions(
+                PublicKeyCredentialCreationOptions(
```

Comment:
Giving this a try, and can't see why the following would get raised
Wondering if the utility could have a better error raised to indicate the field/dict causing the issue
only `int` type I see around is the timeout

>    options, state = f2_server.register_begin(
  File "/opt/penv/lib/python3.9/site-packages/fido2/server.py", line 194, in register_begin
    PublicKeyCredentialCreationOptions(
  File "<string>", line 12, in __init__
  File "/opt/penv/lib/python3.9/site-packages/fido2/utils.py", line 215, in __post_init__
    value = _parse_value(hints[f.name], value)
  File "/opt/penv/lib/python3.9/site-packages/fido2/utils.py", line 176, in _parse_value
    return [_parse_value(t, v) for v in value]
  File "/opt/penv/lib/python3.9/site-packages/fido2/utils.py", line 176, in <listcomp>
    return [_parse_value(t, v) for v in value]
  File "/opt/penv/lib/python3.9/site-packages/fido2/utils.py", line 199, in _parse_value
    return t.from_dict(value)
  File "/opt/penv/lib/python3.9/site-packages/fido2/utils.py", line 259, in from_dict
    if key in data:
TypeError: argument of type 'int' is not iterable

Reply:
I've pushed some commits adding more descriptive Exceptions here. Please try it out and also let me know what was causing the error in the first place, I'd be curious to know.

Reply:
Thanks @dainnilsson  
Looks like I got caught with a migration issue, previous code was setting `allowed_algorithms`, but using just the list of int rather than `list PublicKeyCredentialParameters` 
ie. `.allowed_algorithms = [ES256.ALGORITHM, RS256.ALGORITHM]`

> .register_begin(
  File "/opt/penv/lib/python3.9/site-packages/fido2/server.py", line 194, in register_begin
    PublicKeyCredentialCreationOptions(
  File "<string>", line 12, in __init__
  File "/opt/penv/lib/python3.9/site-packages/fido2/utils.py", line 218, in __post_init__
    raise ValueError(
ValueError: Error parsing field pub_key_cred_params for PublicKeyCredentialCreationOptions